pkgs/by-name/up/upmpdcli/package.nix

```
@@ -0,0 +1,66 @@
+{
+  fetchgit,
+  stdenv,
+  lib,
+  autoreconfHook,
+  pkg-config,
+  libupnpp,
+  libnpupnp,
+  curl,
+  expat,
+  libmpdclient,
+  libmicrohttpd,
+  jsoncpp,
+  makeWrapper,
+  python3,
+  recoll,
+  mutagen,
+}:
+
+let
+  recoll' = recoll.override { python3Packages = python3.pkgs; };
+  mutagen' = python3.pkgs.mutagen;
+in
+
+stdenv.mkDerivation rec {
+  pname = "upmpdcli";
+  version = "1.9.5";
+
+  src = fetchgit {
+    url = "https://framagit.org/medoc92/upmpdcli.git";
+    rev = "upmpdcli-v${version}";
+    sha256 = "sha256-Ai41slMaJlVnENRevelYdDIR6x4XCjh7l6IVIZbUV+s=";
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+    libupnpp
+    libnpupnp
+    curl
+    expat
+    libmpdclient
+    libmicrohttpd
+    jsoncpp
+    makeWrapper
+    python3
+    recoll'
+  ];
```

You should split these into the dependencies needed for the machine building the program and the ones needed for the machine running the program. Maybe something like this (but I haven't tested, so it might be different). Removed `recoll'`, since it seems to be added when the application gets wrapped.

```suggestion
  nativeBuildInputs = [
    autoreconfHook
    pkg-config
    makeWrapper
    python3
  ];

  buildInputs = [
    jsoncpp
    libmpdclient
    libmicrohttpd
    libupnpp
    libnpupnp
    expat
    curl
  ];
```